docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md

```diff
@@ -0,0 +1,64 @@
+- Implement a query builder. This is highly useful for users who are not familiar with the query language of the data-source. See, for example, the [Query Builder for Microsoft SQL Server](https://grafana.com/docs/grafana/latest/datasources/mssql/query-editor/#builder-mode) which helps write SQL queries for that service.
+- Add a health check for your plugin. This is used to verify that the data-source is working properly. More information in [Health checks](/introduction/backend-plugins#health-checks). Depending on the type of plugin (frontend or backend) the implementation methods differ. See this [example](https://github.com/grafana/grafana-plugin-examples/blob/5441fe2f818e28cdeb06eb7066ff198dd34bb0ab/examples/datasource-http/src/DataSource.ts#L81-L115) for a health check in the frontend or this [example](https://github.com/grafana/grafana-plugin-examples/blob/0532f8b23645251997088ac7a1707a72d3fd9248/examples/datasource-basic/pkg/plugin/datasource.go#L77) in the backend.
+- Add [dashboard variables](https://grafana.com/docs/grafana/latest/dashboards/variables/) support. Dashboard (or template) variables allow users to create dynamic dashboards. There are two aspects of adding support for variables in your plugin. The first is allowing querying your data-source and return values that will be used as variables. The second is replacing existing variables in other queries. See how you can do it [here](/create-a-plugin/extend-a-plugin/add-support-for-variables#add-support-for-query-variables-to-your-data-source). Pay special attention when selecting "All values" since it may require specific logic to join variable values.
+- Add annotations support. Annotations allow users to add contextual information to their dashboards and it's possible to use queries to define them. See how you can do it [here](/create-a-plugin/extend-a-plugin/enable-for-annotations).
+- When building frontend components, make sure to use [Grafana components](https://developers.grafana.com/ui/latest/index.html?path=/docs/docs-overview-intro--page) as the base and follow the [Saga Design System](https://grafana.com/blog/2023/11/07/saga-design-system-shaping-the-future-of-user-experiences-at-grafana-labs/).
+- Add query editor help. Query editors can be complex and it's useful to provide help to users. See how you can do it [here](/create-a-plugin/extend-a-plugin/add-query-editor-help).
+- Skip hidden or empty queries. This avoids executing unnecessary or wrong queries. See this [example](https://github.com/grafana/grafana/blob/fd5f66083c91b9759ae7772f99b80c9342b93290/public/app/plugins/datasource/loki/datasource.ts#L1085).
+- Specify a default query. This can be useful for users to discover how queries are written for the plugin. See an [example](https://github.com/grafana/grafana-plugin-examples/blob/0532f8b/examples/datasource-streaming-backend-websocket/streaming-backend-websocket-plugin/src/datasource.ts#L39-L41).
+- Avoid using `console.log` in your plugin.
```

Comment:
why?

Reply:
it's a best practice, in general: https://eslint.org/docs/latest/rules/no-console. It can clutter the browser console and hide real errors.